src/version.rs

```diff
@@ -0,0 +1,68 @@
+use super::*;
+use crate::request::Asset;
+
+/// Version of a project
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct Version<ID> {
+    /// The ID of the version
+    pub id: ID,
+    /// The Display name of the version
+    pub name: String,
+    /// The identifier of the version
+    pub identifier: String,
+    /// The ID of the parent project
+    pub project_id: ID,
+    /// Jar files of the Mod
+    pub files: Vec<Asset>,
+    /// Downloads of this version
+    pub downloads: usize,
+    /// Loaders this version supports
+    pub loaders: Vec<ModLoader>,
+    /// Game versions this version supports
+    pub game_versions: Vec<String>,
+    /// When this version was published
+    pub published: Datetime,
+    /// The type of version, Release, Beta or Alpha
+    pub version_type: VersionType,
+    /// Dependencies on other projects
+    pub dependencies: Vec<VersionDependency<ID>>,
+}
+
+/// Dependency on a Project's version
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct VersionDependency<ID> {
+    /// The ID of the project to depend on
+    pub project_id: Option<ID>,
```

Comment:
What relation is there between the project_id and the version?
If it is exclusive (one **OR** the other), an enum `VersionDescriptor` could be interesting to make it explicit, and if both can / must be defined, then this system works finely

Reply:
On Modrinth, they're exclusive but on CurseForge they're not. CurseForge does this extremely annoying thing where you cannot use a file id independently, you _have_ to include the project id with it.

This is also why the version calls have rather confusing parameters, because both are required (but only for cf)